content/06_Transformation.md

```diff
@@ -0,0 +1,372 @@
+
+These two cases—fanning out then back in, and removing or avoiding a layer of observables of observables—come up quite often, which makes `SelectMany` an important method. (It's not surprising that I was unable to avoid using it in earlier examples.)
+
+As it happens, `SelectMany` is also a particularly important operator in the mathematical theory that Rx is based on. It is a fundamental operator, in the sense that it is possible to build many other Rx operators with it. [Section XXX in Appendix C](./C_AlgebraicUnderpinnings) shows how you can implement `Select` and `Where` using `SelectMany`.
+
+## Cast
+
+C#'s type system is not omniscient. Sometimes we might know something about the type of the values emerging from an observable source that is not reflected in that source's type. This might be based on domain-specific knowledge. For example, with the AIS message broadcast by ships, we might know that if the message type is 3, it will contain navigation information. That means we could write this:
+
+```cs
+IObservable<IVesselNavigation> type3 = receiverHost.Messages
+    .Where(v => v.MessageType == 3)
+    .Cast<IVesselNavigation>();
+```
+
+This uses `Cast`, a standard LINQ operator that we can use whenever we know that the items in some collection are of some more specific type than the type system ahs been able to deduce.
+
+The difference between `Cast` and the [`OfType` operator shown in Chapter 5](./05_Filtering.md#oftype) is the way in which they handle items that are not of the specified type. `OfType` is a filtering operator, so it just filters out any items that are not of the specified type. But with `Cast` (as with a normal C# cast expression) we are asserting that we expect the source items to be of the specified type, so the observable returned by `Cast` will invoke its subscriber's `OnError` if its source produces an item that is not compatible with the specified type.
+
+This distinction might be easier to see if we recreate the functionality of `Cast` and `OfType` using other more fundamental operators.
+
+```csharp
+// source.Cast<int>(); is equivalent to
+source.Select(i => (int)i);
+
+// source.OfType<int>();
+source.Where(i=> i is int).Select(i=>(int)i);
+```
+
+## Materialize and Dematerialize			
```

Comment:
Having read this section, I still don't understand the use cases where I would use this? Are there any real world examples you could use to bring this to life?

Reply:
Again, you've highlighted some lines that span two sections. Are you referring to `Cast` (which is the section lines 302-304 are in) or *Materialize and Dematerialize* (which is what the final selected line is in)?

I'm guessing the latter, but please let me know if that guess is wrong.

Reply:
I think it was Materialize and Dematerialize